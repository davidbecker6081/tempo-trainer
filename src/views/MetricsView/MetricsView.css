.metrics-view-container {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 4em;
  padding: 0 2rem;
  width: 60vw;
  background-color: #72C6E9;
}

.metrics-view-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}
