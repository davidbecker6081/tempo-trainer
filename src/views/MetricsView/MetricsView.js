import React from 'react';
import PropTypes from 'prop-types';
import MetricTabs from '../../components/MetricTabs/MetricTabs';
import Graph from '../../components/Graph/Graph';
import RangeSlider from '../../components/RangeSlider/RangeSlider';
import './MetricsView.css';

const MetricsView = ({
  range,
  handleRangeChange,
  dataHelper,
  channelSet,
  handleChannelChange,
  originalRange,
}) =>
  (
    <section className="container metrics-view">
      <h2>Metrics</h2>
      <article className="metrics-view-wrapper">
        <MetricTabs
          handleChannelChange={handleChannelChange}
          channelSet={channelSet}
        />
        <Graph
          range={range}
          dataHelper={dataHelper}
          channelSet={channelSet}
        />
        <RangeSlider
          range={range}
          handleRangeChange={handleRangeChange}
          originalRange={originalRange}
        />
      </article>
    </section>
  );

export default MetricsView;

MetricsView.propTypes = {
  channelSet: PropTypes.string.isRequired,
  dataHelper: PropTypes.shape({
    originalData: PropTypes.shape({}),
    minuteData: PropTypes.arrayOf(PropTypes.shape({})),
    channels: PropTypes.arrayOf(PropTypes.string),
    GPSCoords: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
  range: PropTypes.arrayOf(PropTypes.number).isRequired,
  handleRangeChange: PropTypes.func.isRequired,
  handleChannelChange: PropTypes.func.isRequired,
  originalRange: PropTypes.arrayOf(PropTypes.number).isRequired,
};
