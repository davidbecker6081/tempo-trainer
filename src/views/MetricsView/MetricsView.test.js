import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import MetricsView from './MetricsView';
import DataHelper from '../../components/DataCleaner/DataCleaner';
import workoutData from '../../__mock__/workout-data.json';

describe('MetricsView', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      originalRange: [],
      dataHelper: {},
      range: [],
      handleRangeChange: () => {},
      channelSet: 'power',
      handleChannelChange: () => {},
    };
  });

  describe('originalRange is defined', () => {
    beforeEach(() => { mockProps.originalRange = [1, 5]; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('range is defined', () => {
    beforeEach(() => { mockProps.range = [0, 5]; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('dataHelper is defined', () => {
    beforeEach(() => { mockProps.dataHelper = new DataHelper(workoutData); });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('channelSet is defined', () => {
    beforeEach(() => { mockProps.channelSet = 'power'; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
