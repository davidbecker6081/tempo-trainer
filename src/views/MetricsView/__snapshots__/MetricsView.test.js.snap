// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricsView channelSet is defined should render correclty 1`] = `undefined`;

exports[`MetricsView dataHelper is defined should render correclty 1`] = `undefined`;

exports[`MetricsView originalRange is defined should render correclty 1`] = `undefined`;

exports[`MetricsView range is defined should render correclty 1`] = `undefined`;
